<?xml version="1.0"?>

<launch>
	<arg name="robotName" value="bebop"/>
	<!-- SLOW -->
	<!-- arg name="kp" default="0.300"/>
	<arg name="ki" default="0.100"/>
	<arg name="kd" default="0.001"/>
	<arg name="alpha_p" default="0.1"/>
	<arg name="alpha_d" default="0.5"/>
	<arg name="ka" default="2.0"/>
	<arg name="kb" default="1.0"/ -->
	<!-- FAST -->
	<arg name="kp" default="1.2"/>
	<arg name="ki" default="0.001"/>
	<arg name="kd" default="0.4"/>
	<arg name="alpha_p" default="0.5"/>
	<arg name="alpha_d" default="0.5"/>
	<arg name="ka" default="1.5"/>
	<arg name="kb" default="0.0"/>

	<remap from="/uav/command_velocity" to="/$(arg robotName)/command_velocity"/>
	<remap from="/uav/command/motor_speed" to="/$(arg robotName)/command/motor_speed"/>
	<remap from="/uav/position" to="/$(arg robotName)/position"/>
	<remap from="/uav/velocity" to="/$(arg robotName)/linear_velocity"/>
	<remap from="/uav/odometry" to="/$(arg robotName)/odometry"/>
	<remap from="/uav/true_odometry" to="/$(arg robotName)/ground_truth/odometry"/>
	<remap from="/uav/trajectory" to="/$(arg robotName)/trajectory"/>
	<remap from="/uav/trajectory_velocity" to="/$(arg robotName)/trajectory_velocity"/>

	<node name="dit2flc" pkg="controllers" type="dit2flc" output="screen" args="$(arg kp) $(arg ki) $(arg kd) $(arg ka) $(arg kb) $(arg alpha_p) $(arg alpha_d)"/>
	<node name="trajectory" pkg="controllers" type="trajectory" output="screen" args=""/>

	<!-- node name="dynamic_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" args=""/ -->
</launch>
